utils/process.py

```
@@ -78,61 +88,320 @@
+                logger.warning("Failed to load security.process.logging.log_command_execution from config")
+        return True
+
+
+# Global security config instance
+_security_config = SecurityConfig()
+
+
+def _ensure_list_command(command: list[str] | tuple[str, ...]) -> None:
+    """Validate that command is a non-empty list[str] or tuple[str, ...] of strings."""
+    if not isinstance(command, (list, tuple)):
         raise TypeError("command must be a list[str] (or tuple[str])")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)